editor/editor_themes.cpp

```diff
@@ -233,37 +248,50 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
 	int LIGHT_COLOR = 2;
 	bool dark_theme = (icon_font_color_setting == AUTO_COLOR && ((base_color.r + base_color.g + base_color.b) / 3.0) < 0.5) || icon_font_color_setting == LIGHT_COLOR;
 
-	Color dark_color_1 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast);
-	Color dark_color_2 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast * 1.5);
-	Color dark_color_3 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast * 2);
+	const Color dark_color_1 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast);
+	const Color dark_color_2 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast * 1.5);
+	const Color dark_color_3 = base_color.linear_interpolate(Color(0, 0, 0, 1), contrast * 2);
 
-	Color contrast_color_1 = base_color.linear_interpolate((dark_theme ? Color(1, 1, 1, 1) : Color(0, 0, 0, 1)), MAX(contrast, default_contrast));
-	Color contrast_color_2 = base_color.linear_interpolate((dark_theme ? Color(1, 1, 1, 1) : Color(0, 0, 0, 1)), MAX(contrast * 1.5, default_contrast * 1.5));
+	const Color background_color = dark_color_2;
 
-	Color font_color = dark_theme ? Color(1, 1, 1) : Color(0, 0, 0);
-	Color font_color_disabled = dark_theme ? Color(0.6, 0.6, 0.6) : Color(0.45, 0.45, 0.45);
+	// white (dark theme) or black (light theme), will be used to generate the rest of the colors
+	const Color mono_color = dark_theme ? Color(1, 1, 1) : Color(0, 0, 0);
 
-	Color separator_color = dark_theme ? Color(1, 1, 1, 0.1) : Color(0, 0, 0, 0.1);
+	const Color contrast_color_1 = base_color.linear_interpolate(mono_color, MAX(contrast, default_contrast));
+	const Color contrast_color_2 = base_color.linear_interpolate(mono_color, MAX(contrast * 1.5, default_contrast * 1.5));
 
-	Color tab_color = highlight_tabs ? base_color.linear_interpolate(font_color, contrast) : base_color;
-	const int border_width = CLAMP(border_size, 0, 3) * EDSCALE;
+	const Color font_color = mono_color.linear_interpolate(base_color, 0.25);
+	const Color font_color_hl = mono_color.linear_interpolate(base_color, 0.15);
+	const Color font_color_disabled = Color(mono_color.r, mono_color.g, mono_color.b, 0.3);
+	const Color color_disabled = mono_color.inverted().linear_interpolate(base_color, 0.7);
+	const Color color_disabled_bg = mono_color.inverted().linear_interpolate(base_color, 0.9);
+
+	const Color alpha_1 = Color(mono_color.r, mono_color.g, mono_color.b, 0.1);
+	const Color alpha_2 = Color(mono_color.r, mono_color.g, mono_color.b, 0.1);
+	const Color alpha_3 = Color(mono_color.r, mono_color.g, mono_color.b, 0.1);
```

Comment:
they are all the same is that intended?

Reply:
oh forgot to remove those

Reply:
@toger5 i wanted to have generalized alpha values, but there are too much variation all over the editor

Reply:
Maybe it would be usefult to have a function `Color Color::with_alpha(float p_alpha)`
So that u at least dont need additional lines to declare those different values